File: frontend/src/types/notification.ts
/**
 * Notification types - with intentional issues
 * 
 * MNT: 'any' type abuse
 */

// MNT: Using 'any' instead of proper types
export interface Notification {
  id?: any
  userId?: any
  type?: any
  recipient?: any
  subject?: any
  content?: any  // SEC: Could contain XSS
  status?: any
  priority?: any
  channel?: any
  retryCount?: any
  createdAt?: any
  sentAt?: any
  errorMessage?: any
  metadata?: any
}

// MNT: Another 'any' type
export interface NotificationPreferences {
  id?: any
  userId?: any
  emailEnabled?: any
  smsEnabled?: any
  pushEnabled?: any
  orderConfirmation?: any
  orderShipped?: any
  orderDelivered?: any
  promotional?: any
}

// MNT: 'any' type abuse
export interface WebhookConfiguration {
  id?: any
  name?: any
  url?: any  // SEC: SSRF vulnerability
  eventTypes?: any
  active?: any
  retryCount?: any
  timeoutSeconds?: any
}

// MNT: Generic 'any' type
export type NotificationType = any  // Should be: 'email' | 'sms' | 'push'

// MNT: Another generic type
export type NotificationStatus = any  // Should be: 'pending' | 'sent' | 'failed'

// TODO: Define proper types
// FIXME: Replace 'any' with proper types

File: backend/src/main/java/com/sonarshowcase/model/Notification.java
package com.sonarshowcase.model;

import jakarta.persistence.*;
import java.util.Date;

/**
 * Notification entity - represents a notification sent to a user
 * 
 * @author SonarShowcase
 */
@Entity
@Table(name = "notifications")
public class Notification {
    
    /**
     * Default constructor for Notification.
     */
    public Notification() {
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "user_id")
    private Long userId;
    
    private String type; // MNT: Should be enum (email, sms, push)
    
    private String recipient;
    
    private String subject;
    
    @Column(length = 5000)
    private String content; // SEC: Could contain XSS
    
    private String status; // MNT: Should be enum (pending, sent, failed)
    
    private String priority; // MNT: Should be enum
    
    private String channel; // MNT: Should be enum
    
    @Column(name = "retry_count")
    private Integer retryCount;
    
    @Column(name = "created_at")
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;
    
    @Column(name = "sent_at")
    @Temporal(TemporalType.TIMESTAMP)
    private Date sentAt;
    
    @Column(name = "error_message")
    private String errorMessage;

    // Getters and Setters
    
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getRecipient() {
        return recipient;
    }

    public void setRecipient(String recipient) {
        this.recipient = recipient;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getPriority() {
        return priority;
    }

    public void setPriority(String priority) {
        this.priority = priority;
    }

    public String getChannel() {
        return channel;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public Integer getRetryCount() {
        return retryCount;
    }

    public void setRetryCount(Integer retryCount) {
        this.retryCount = retryCount;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getSentAt() {
        return sentAt;
    }

    public void setSentAt(Date sentAt) {
        this.sentAt = sentAt;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

